ete4/smartview/renderer/gardening.pyx

```diff
@@ -16,7 +16,79 @@ def sort(tree, key=None, reverse=False):
     tree.children.sort(key=key, reverse=reverse)
 
 
-def root_at(node, bprops=None):
+def root_at(outgroup, bprops=None):
+    """ set the node "outgroup" as the first child of its current tree root """
+    import copy
+
+    t = outgroup.root
+    assert outgroup != t, 'cannot set the absolute tree root as an outgroup'
+
+    if len(t.children) > 2:
+        new_node = t.__class__()
+        for ch in list(t.children[1:]):
+            t.children.remove(ch)
+            new_node.children.append(ch)
+            ch.up = new_node
+
+        for bprop in ['support'] + (bprops or []):
+            if t.children[0].props.get(bprop, None) is not None:
+                new_node.props[bprop] = t.children[0].props[bprop]
+        t.children.append(new_node)
+        new_node.up = t
+
+
+    # if node is already a direct child of the root node, we  just make sure the
+    # outgroup is the "first" child and branch distances are balanced
+    if outgroup.up == t:
+        t.children.remove(outgroup)
+        t.children.insert(0, outgroup)
+
+        # balance distances
+        total_dist = (t.children[0].dist or 0.0) + (t.children[1].dist or 0.0)
+        t.children[0].dist = total_dist / 2.0
+        t.children[1].dist = total_dist / 2.0
+
+        return t
+
+    # if the outgroup is somewhere down in the tree structure, we want to rehang
+    # all branches but keep keep the original absolute tree root instance
+    # intact.
+    #
+    # So let's first detach current tree root from the tree and create another node
+    # instance (intermediate) that replaces it in the topology
+    intermediate_node = t.__class__()
+    intermediate_node.name = "inter"
+    for ch in t.children:
+        intermediate_node.add_child(ch)
+    t.children = []
+    t.up = None
+
+    # transfer branch properties from the original root node to the new
+    # intermediate node We just want those props to be transferred, the rest
+    # should stay in tree node.
+    #if bprops:
+    #    for p in bprops:
+    #        if p in t.props:
+    #            intermediate_node.props[p] = copy.deepcopy(t.props[p])
+
+    # Now, let's place the original absolute tree root instance in the branch
+    # that will become the new root, spliting the branch into two
+    split_branch(outgroup, bprops, t)
+
+    # and then rehang all nodes above the new outgroup, tranferring bprops
+    # appropriately
+    old_root, node_id = get_root_id(t)
+    root = old_root  # current root, which will change in each iteration
+    for child_pos in node_id:
+        root = rehang(root, child_pos, bprops)
+
+    # clean up topology to avoid a single-child node derived from the old root
+    if len(old_root.children) == 1:
+        join_branch(old_root)
+
+    return root
+
+def root_at_orig(node, bprops=None):
```

Comment:
If we want to change to the new `root_at()`, I think that it doesn't make sense to keep `root_at_orig()`. It's certainly not referenced from anywhere, and I think it would better be removed in this case.